bundles/org.smarthomej.binding.viessmann/src/main/java/org/smarthomej/binding/viessmann/internal/handler/ViessmannBridgeHandler.java

```diff
@@ -0,0 +1,317 @@
+                        devicesList.add(deviceId);
+                    }
+                    logger.trace("Device ID: {}, Type: {}", deviceId, deviceType);
+                }
+            }
+        }
+    }
+
+    public boolean setData(@Nullable String url, @Nullable String json) {
+        if (url != null && json != null) {
+            countApiCalls();
+            return api.setData(url, json);
+        }
+        return false;
+    }
+
+    private Integer getPollingInterval() {
+        if (this.config.pollingInterval > 0) {
+            return this.config.pollingInterval;
+        } else {
+            Integer interval = (86400 / (this.config.apiCallLimit - this.config.bufferApiCommands) * devicesList.size())
+                    + 1;
+            return interval;
+        }
+    }
+
+    private void countApiCalls() {
+        apiCalls++;
+        updateState(COUNT_API_CALLS, DecimalType.valueOf(Integer.toString(apiCalls)));
+    }
+
+    private void checkResetApiCalls() {
+        LocalTime time = LocalTime.now();
+        if (time.isAfter(LocalTime.of(00, 00, 01)) && (time.isBefore(LocalTime.of(01, 00, 00)))) {
+            if (countReset) {
+                logger.debug("Resettig API Call counts");
+                apiCalls = 0;
+                countReset = false;
+            }
+        } else {
+            countReset = true;
+        }
+    }
+
+    private void pollingFeatures() {
+        List<String> devices = pollingDevicesList;
+        if (devices != null) {
+            for (String deviceId : devices) {
+                logger.debug("Loading Featueres from Device ID: {}", deviceId);
+                getAllFeaturesByDeviceId(deviceId);
+            }
+        }
+    }
+
+    public void getAllFeaturesByDeviceId(String deviceId) {
+        try {
+            FeaturesDTO allFeatures = api.getAllFeatures(deviceId);
+            countApiCalls();
+            if (allFeatures != null) {
+                List<FeatureDataDTO> featuresData = allFeatures.data;
+                for (FeatureDataDTO featureDataDTO : featuresData) {
+                    notifyChildHandlers(featureDataDTO);
+                }
+            }
+        } catch (JsonSyntaxException | IllegalStateException e) {
+            logger.warn("Parsing Viessmann response fails: {}", e);
+        }
+    }
+
+    private void startViessmannBridgePolling(Integer pollingIntervalS) {
+        ScheduledFuture<?> currentPollingJob = viessmannBridgePollingJob;
+        if (currentPollingJob == null) {
+            viessmannBridgePollingJob = scheduler.scheduleWithFixedDelay(() -> {
+                logger.debug("Refresh job scheduled to run every {} seconds for '{}'", pollingIntervalS,
+                        getThing().getUID());
+                api.checkExpiringToken();
+                checkResetApiCalls();
+                pollingFeatures();
+            }, 1, TimeUnit.SECONDS.toSeconds(pollingIntervalS), TimeUnit.SECONDS);
```

Comment:
```suggestion
            }, 1, pollingIntervalS, TimeUnit.SECONDS);
```

no need to convert. Also below